File: lincoln/ruby/lib/lang/plan/clause.rb
class Clause 
  include Comparable
  def initialize(location)
		@location = location
	end
	
	attr_reader :location

	def <=>(o)
		return location.<=>(o.location)
	end

	def to_s
	  raise "Clause.to_s needs to be subclassed"
  end
	
	def set(program) 
	  raise "Clause.set needs to be subclassed"
  end
}

File: lincoln/java/src/jol/lang/plan/Clause.java
package p2.lang.plan;

import p2.core.Runtime;
import p2.types.exception.UpdateException;

public abstract class Clause implements Comparable<Clause> {
	private xtc.tree.Location location;
	
	protected Clause(xtc.tree.Location location) {
		this.location = location;
	}
	
	public xtc.tree.Location location() {
		return this.location;
	}
	
	public int compareTo(Clause o) {
		return location.compareTo(o.location);
	}

	@Override
	public abstract String toString();
	
	public abstract void set(Runtime context, String program) throws UpdateException;

}
